docs/rules/no-extraneous-dependencies.md

```diff
@@ -17,6 +17,22 @@ This rule supports the following options:
 
 `bundledDependencies`: If set to `false`, then the rule will show an error when `bundledDependencies` are imported. Defaults to `true`.
 
+`considerInParents`: If set to `true`, then the rule will take `package.json` files in parent directories into account for
+all dependency types, unless overridden by one of the other related options.
+
+`considerDevInParents`: If set to `true`, then the rule will take `package.json` files in parent directories into account for
+`devDependencies`. If set to `false`, it will override `considerInParents` for `devDependencies`.
+
+`considerOptionalInParents`: If set to `true`, then the rule will take `package.json` files in parent directories into account for
+`optionalDependencies`. If set to `false`, it will override `considerInParents` for `optionalDependencies`.
+
+`considerPeerInParents`: If set to `true`, then the rule will take `package.json` files in parent directories into account for
+`peerDependencies`. If set to `false`, it will override `considerInParents` for `peerDependencies`.
+
+`considerBundledInParents`: If set to `true`, then the rule will take `package.json` files in parent directories into account for
+`bundledDependencies`. If set to `false`, it will override `considerInParents` for `bundledDependencies`.
```

Comment:
oof, this is a lot of new schema options to add. maybe `considerInParents` takes an array which must have 1-5 of "prod", "dev", "peer", "optional", "bundled" in it?